src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -2666,33 +2843,317 @@ static class E
+            Assert.Equal("void C.M<System.Object?>()", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+            Assert.Equal(new[] { "void C.M<System.Object?>()" }, model.GetMemberGroup(memberAccess).ToTestDisplayStrings());
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69222")]
+        public void MethodGroup_GenericExtensionMethod_Constraint_ImplicitTypeArguments()
+        {
+            var source = """
+var x = new object().M;
+x();
+
+static class E1
+{
+    public static void M<T>(this T t)
+    {
+        System.Console.Write("ran");
+    }
+}
+
+static class E2
+{
+    public static void M<T>(this T t, object ignored) where T : struct { }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            // ILVerify: Unrecognized arguments for delegate .ctor.
+            CompileAndVerify(comp, expectedOutput: "ran", verify: Verification.FailsILVerify);
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var memberAccess = GetSyntax<MemberAccessExpressionSyntax>(tree, "new object().M");
+            Assert.Equal("void System.Object.M<System.Object>()", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+            Assert.Equal(new[] { "void System.Object.M<System.Object>()" }, model.GetMemberGroup(memberAccess).ToTestDisplayStrings());
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69222")]
+        public void MethodGroup_GenericExtensionMethod_Constraint_ExplicitTypeArguments()
+        {
+            var source = """
+var x = new object().M<object>;
+x();
+
+static class E1
+{
+    public static void M<T>(this T t)
+    {
+        System.Console.Write("ran");
+    }
+}
+
+static class E2
+{
+    public static void M<T>(this T t, object ignored) where T : struct { }
+}
+""";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            // ILVerify: Unrecognized arguments for delegate .ctor.
+            CompileAndVerify(comp, expectedOutput: "ran", verify: Verification.FailsILVerify);
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var memberAccess = GetSyntax<MemberAccessExpressionSyntax>(tree, "new object().M<object>");
+            Assert.Equal("void System.Object.M<System.Object>()", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+
+            Assert.Equal(new[] { "void System.Object.M<System.Object>()", "void System.Object.M<System.Object>(System.Object ignored)" },
+                model.GetMemberGroup(memberAccess).ToTestDisplayStrings());
+        }
+
+        [Fact]
+        public void GenericExtensionMethod_ArityCountsInSignature()
+        {
+            // Arity counts as part of "signature" so we have two different signatures even after reducing extension methods
```

Comment:
>// Arity counts as part of "signature" so we have two different signatures even after reducing extension methods [](http://example.com/codeflow?start=12&length=112)

This doesn't feel right. Since all type arguments are supplied, they shouldn't matter.